core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -54,10 +52,10 @@
     public static final ParseField FUZZY_REWRITE_FIELD = new ParseField("fuzzy_rewrite");
     public static final ParseField MINIMUM_SHOULD_MATCH_FIELD = new ParseField("minimum_should_match");
     public static final ParseField OPERATOR_FIELD = new ParseField("operator");
-    public static final ParseField MAX_EXPANSIONS_FIELD = new ParseField("max_expansions");
+    public static final ParseField MAX_EXPANSIONS_FIELD = new ParseField("max_expansions").withAllDeprecated("match_phrase_prefix query");
```

Comment:
max_expansions may still be used along with fuzziness, so it may not be ok to deprecate. Can you double check? This is why splitting MatchQuery would help. It is hard to figure out what happens when.